inspire_disambiguation/core/es/readers.py

```diff
@@ -0,0 +1,181 @@
+    if signature_block:
+        partial_authors_query += Q(
+            "term", authors__signature_block__raw=signature_block
+        )
+    authors_query = Q("nested", path="authors", query=partial_authors_query)
+    query = (
+        LiteratureSearch()
+        .query(Q("bool", must=[literature_query, authors_query]))
+        .params(size=conf["ES_MAX_QUERY_SIZE"], _source=SIGNATURE_FIELDS)
+    )
+    return query
+
+
+def get_signatures(signature_block=None, only_curated=False):
+    """Get all signatures from ES.
+
+    Args:
+        signature_block (str, optional): string representation of selected signature
+            block. If empty or None returns all signature blocks. Defaults to None.
+        only_curated (bool, optional): If set to true return only curated signatures,
+            otherwise return all. Defaults to False.
+
+    Returns:
+        list: dicts of signatures.
+
+    Example:
+        >>> get_signatures("ABDa", True)
+        [Signature(...), Signature(...), Signature(...)]
+
+    """
+    query = get_lit_records_query(signature_block, only_curated)
+    results = []
+    for record in query.scan():
+        record = record.to_dict()
+        for author in record.get("authors"):
+            if only_curated and not author.get("curated_relation"):
+                continue
+            if signature_block and author.get("signature_block") != signature_block:
+                continue
+            signature = Signature.build(author, record)
+            # Skip signatures marked as curated which do not have author_id
+            if only_curated and not signature.get("author_id"):
+                continue
+            results.append(signature)
+    return results
+
+
+def get_input_clusters(signatures):
+    """Return input clusters for given signatures.
+
+    Args:
+        signatures (iterable): Iterable which contains dicts of signatures
+            for which input_clusters should be built.
+
+    Returns:
+        list: list of all input clusters for provided signatures.
+
+    Example:
+        >>> signatures = get_signatures('ABC')
+        >>> get_input_clusters(signatures)
+        [{'author_id': 1234,
+          'cluster_id': 0,
+          'signature_uuids': [
+            'd83736ab-0672-46ac-9efa-065c685bf26e',
+            'cfcd2ea7-d1ce-4d76-a733-b3d1c5e2cb11'
+          ]},
+          {'author_id
+
+           ]
+
+    """
+    input_clusters = []
+    signatures_with_author = defaultdict(list)
+    signatures_without_author = []
+    for signature in signatures:
+        if signature.get("is_curated_author_id"):
+            signatures_with_author[signature["author_id"]].append(
+                signature["signature_uuid"]
+            )
+        else:
+            signatures_without_author.append(signature["signature_uuid"])
+    for cluster_id, (author_id, signature_uuids) in enumerate(
+        six.iteritems(signatures_with_author)
```

Comment:
aren't we using python3 only? why we need ``six``?

Reply:
Yup, you are right. I forgot to update this old code.